### ISSUE TITLE
Not reproducible of table 1,3 results.

### ISSUE THREAD

**Issue by kunaljangid01** [psandovalsegura/learn-from-unlearnable#1]
The results of Table 1 and Table 3 are not reproducible, as there are changes of more than a 4-5%  in the results.
Can you help me with this and provide the correct saved checkpoints and the saved models and dataset of svhn, cifar100, and imagenet; and for the different models as VGG-16, GoogleNet, Vit to reproduce the additional tables results.                        
In Table 3, the reproducible results of Adversarial poisoning and AR (l2) are quite higher, approximately 20 - 25%. Is this correct, or may I have made a mistake in this?

**Comment by psandovalsegura** [comment]
Can you provide a specific row/number of Table 1 or Table 3? Both use CIFAR-10, so it should be easiest to see what the issue is if we focus on one number first.

In Table 1 models are first trained from scratch, checkpoints are saved throughout training, then those checkpoints are finetuned (the last linear layer) to get the DFR numbers. Pretrained models from the web aren't used. Are you using the [ResNet-18 class](https://github.com/psandovalsegura/learn-from-unlearnable/blob/866d1dab6420bb1ea99796c8a263f3b0562e23e9/models/resnet.py#L108)?

CIFAR-100, SVHN, and STL L2 AR poisons can be found here: https://drive.google.com/drive/u/1/folders/1ze0cKAXNcPRkC0TMmObMj-g7Gspp1DpL
and can be loaded with [TAPFormatPoison](https://github.com/psandovalsegura/learn-from-unlearnable/blob/866d1dab6420bb1ea99796c8a263f3b0562e23e9/datasets.py#L62).

**Comment by kunaljangid01** [comment]
Yes,  I am using the ResNet-18 model.  In Table 1, all the MAX DFR TEST Accuracy values are reproduced less than 4-7%.
In table 3, Adversarial Poisoning (error-max) and AR l2 (l2-ar)  ortho proj accuracy is increased by around 25%. Also, Clean, NTGA, and Regions-4  ortho proj accuracy decreased by 2-3% in reproduce codes. 
Also, need to reproduce the additional results from Table 4 to Table 9. Can you guide in this and please provide the necessary datasets for all.

**Comment by psandovalsegura** [comment]
> In Table 1, all the MAX DFR TEST Accuracy values are reproduced less than 4-7%

What is the max DFR value you're getting? and for what row (poison)? 

I recommend using the AR(L2) CIFAR 10 poison to debug what's wrong on your end. Paper reports ~70% max DFR accuracy for AR(L2). What is the value you're getting? and can you plot for me your equivalent of Figure 2 (c) where you show the DFR accuracy over time for different checkpoints of AR(L2)?

**Comment by kunaljangid01** [comment]
![Image](https://github.com/user-attachments/assets/0887cded-73e5-4b8e-90db-c9a97730444a)


This is the Figure 2 (c) curve as I plot it AR vs DFR AR on Cifar-10 Test Accuracy.
"Paper reports ~70% max DFR accuracy for AR(L2)." You mention this, but in Table 1, 58.73% Max DFR accuracy for AR(L2). 
I get **57.96% Max DFR accuracy for AR(L2)** and **64.98, 52.49, 36.35, 41.56, and 35.69% DFR accuracy for Adversarial Poisoning, NTGA, Robust Unlearnable, LSP, and OPS+EM, respectively**. I get a decrement in the reproduction results. Can you provide any solution and reason why this is happening?

For Table 3, some results for the Ortho proj are **86.82, 64.90, 54.21, 46.38, 78.53, 35.11, 87.65, 46.50, 87.26, 90.60, and 84.15 for clean, Unlearnable Examples (error-min), Adversarial Poisoning (error-max), AR l2, NTGA, Robust Unlearnable, LSP, OPS+EM, OPS, Unlearnable Examples (error-min-CW), and Regions-4 respectively.**  Variation as **-4%** in clean, -1% Unlearnable Examples (error-min), +**40**% in Adversarial Poisoning (error-max), +**33**% in AR l2, **-4%** in NTGA, **+10%** in Robust Unlearnable, similar in  LSP, **+7%** in  OPS+EM, similar in OPS, +1% in Unlearnable Examples (error-min-CW), and -2% in Regions-4  How this much variation, I will get (particularly in  Adversarial Poisoning (error-max), AR l2, Robust Unlearnable, and OPS+EM)?  


I am running the code you provided using the instructions in the README file.


Is there any change in the provided code?

**Comment by psandovalsegura** [comment]
Your reproduction of Figure 2(c) looks correct for early epochs, but when your DFR AR (red dotted) drops to the same level as the blue curve that indicates something is off because finetuning on a little clean data should always help. One reason for the difference could be in your random selection of a subset of 5000 clean CIFAR-10 training samples

>"Paper reports ~70% max DFR accuracy for AR(L2)."

Table 1, AR(L2) shows Max DFR test accuracy of 58.7% which is within 1% of your 57.96%. 

> I get a decrement in the reproduction results. Can you provide any solution and reason why this is happening?

Likely the randomness in the [selection of the clean images](https://github.com/psandovalsegura/learn-from-unlearnable/blob/866d1dab6420bb1ea99796c8a263f3b0562e23e9/dfr_step_2.py#L29C53-L29C66) used for DFR training (10% of original CIFAR-10 train set = 5000 images). Are you using [the indices in the repo](https://github.com/psandovalsegura/learn-from-unlearnable/tree/main/dfr_indices/CIFAR10)? If so, I'm not sure what's wrong, but I would start by debugging why DFR on a checkpoint is getting you the same accuracy as the original checkpoint. It should always be at least a little better because you're using clean data.

> How this much variation, I will get

There will be variation. I think your numbers are correct/expected. I should have provided averages over maybe 10 runs. The main point of Table 3 was that ortho proj on class-wise data (LSP, OPS, unlearnable error-min CW, and regions-4) can achieve near clean accuracy, which is still what your reproduction shows.

> Is there any change in the provided code?

There are no new changes in the provided code.

**Comment by kunaljangid01** [comment]
Can you share the files for creating the unlearnable datasets??

**Comment by psandovalsegura** [comment]
All the relevant repos are linked in the README. The links here will take you to their repos. I also used their code to generate the unlearnable datasets: https://github.com/psandovalsegura/learn-from-unlearnable?tab=readme-ov-file#setup-instructions
